### PR TITLE
[frontend] delete question

### LINKED ISSUE #6: [Frontend] Delete Questions
- Enable users to delete questions. 
- Implement delete buttons.

### PR DESCRIPTION
**Pull request by carriezhengjr**
Implement delete button to delete a question.
Remove deleted question from question list.
Fixes #6

### CODE REVIEW

File: package-lock.json
Comment (seelengxd) [inline package-lock.json:1]: can we delete this

File: package.json
Comment (seelengxd) [inline package.json:1]: delete this too

File: frontend/src/components/questions/QuestionDeleteIconButton.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  const handleDelete = (): void => {
+    new QuestionsAPI()
+      .deleteQuestion(questionId)
+      .then(() => {
+        // Call the onDelete callback when the delete is successful
+        onDelete(questionId);
+      })
+      .catch((error) => {
+        // Handle errors (e.g., display an error message)
+        console.error('Error deleting question:', error);
```
Comment (seelengxd) [inline frontend/src/components/questions/QuestionDeleteIconButton.tsx:25]:
```suggestion
        console.error('Error deleting question:', error);
        toast({
          title: 'Question deletion failed.',
          status: 'error',
          duration: 9000,
          isClosable: true,
        });
```

to display something to the user :P

`const toast = useToast()` somewhere

**Review by seelengxd — COMMENTED** [review]
it works! (do we need a confirmation dialog? 👀 )

File: package-lock.json
Comment (carriezhengjr) [reply package-lock.json:1]: Screenshot 2023-09-13 at 00 56 08
Hmmm do I delete like this?

**Review by sltsheryl — APPROVED** [review]
LGTM

**Pull request merged**